src/main/java/org/openstreetmap/atlas/checks/validation/areas/OverlappingAOIPolygonCheck.java

```diff
@@ -0,0 +1,237 @@
+        // so that each filter may be tested separately later.
+        return object instanceof Area && !this.isFlagged(object.getIdentifier())
+                && aoiFiltersTest(object);
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+
+        final Area aoi = (Area) object;
+        final Polygon aoiPolygon = aoi.asPolygon();
+        boolean hasOverlap = false;
+
+        // Set of overlapping area AOIs
+        final Set<Area> overlappingArea = Iterables.stream(object.getAtlas().areasIntersecting(
+                aoiPolygon.bounds(),
+                area -> area.getIdentifier() != aoi.getIdentifier() && area.intersects(aoiPolygon)
+                        && aoiFiltersTest(area) && !this.isFlagged(area.getIdentifier())))
+                .collectToSet();
+
+        final CheckFlag flag = new CheckFlag(this.getTaskIdentifier(object));
+        flag.addObject(object);
+
+        // Test each overlapping AOI to see if it overlaps enough and passes the same AOI filter as
+        // the object
+        for (final Area area : overlappingArea)
+        {
+            if (this.hasMinimumOverlapProportion(aoi.asPolygon(), area.asPolygon()))
+            {
+                if (aoiFiltersTest(object, area))
+                {
+                    flag.addObject(area);
+                    flag.addInstruction(this.getLocalizedInstruction(0, object.getOsmIdentifier(),
+                            area.getOsmIdentifier()));
+                    this.markAsFlagged(area.getIdentifier());
+                    hasOverlap = true;
+                }
+            }
+        }
+
+        if (hasOverlap)
+        {
+            this.markAsFlagged(object.getIdentifier());
+            return Optional.of(flag);
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Uses code from findIntersectionType function in BuildingRoadIntersection
+     * 
+     * @param polygon
+     *            {@link Polygon} to check for intersection
+     * @param otherPolygon
+     *            Another {@link Polygon} to check against for intersection
+     * @return true if the polygon to polygon overlap surface area is greater than minimum
+     */
+
+    private boolean hasMinimumOverlapProportion(final Polygon polygon, final Polygon otherPolygon)
+    {
+
+        Clip clip = null;
+        try
+        {
+            clip = polygon.clip(otherPolygon, Clip.ClipType.AND);
+        }
+        catch (final TopologyException e)
+        {
+            System.out
```

Comment:
This should be replaced with logger. How often does this happen?

Reply:
This appears to occur at a rate of about once per 30 flags.